dll/win32/syssetup/wizard.c

```diff
@@ -411,6 +414,123 @@ static const PRODUCT_OPTION_DATA s_ProductOptionData[] =
+    { L"LowBatteryAlarm", L"%SystemRoot%\\Media\\ReactOS_Battery_Low.wav" },
+    { L"MailBeep", L"%SystemRoot%\\Media\\ReactOS_Notify.wav" },
+    { L"Maximize", L"%SystemRoot%\\Media\\ReactOS_Restore.wav" },
+    { L"MenuCommand", L"%SystemRoot%\\Media\\ReactOS_Menu_Command.wav" },
+    { L"MenuPopup", L"" },
+    { L"Minimize", L"%SystemRoot%\\Media\\ReactOS_Minimize.wav" },
+    { L"Open", L"" },
+    { L"PrintComplete", L"%SystemRoot%\\Media\\ReactOS_Print_Complete.wav" },
+    { L"RestoreDown", L"" },
+    { L"RestoreUp", L"" },
+    { L"SystemAsterisk", L"%SystemRoot%\\Media\\ReactOS_Ding.wav" },
+    { L"SystemExclamation", L"%SystemRoot%\\Media\\ReactOS_Exclamation.wav" },
+    { L"SystemExit", L"%SystemRoot%\\Media\\ReactOS_Shutdown.wav" },
+    { L"SystemHand", L"%SystemRoot%\\Media\\ReactOS_Critical_Stop.wav" },
+    { L"SystemNotification", L"%SystemRoot%\\Media\\ReactOS_Balloon.wav" },
+    { L"SystemQuestion", L"%SystemRoot%\\Media\\ReactOS_Ding.wav" },
+    { L"SystemStart", L"%SystemRoot%\\Media\\ReactOS_Startup.wav" },
+    { L"WindowsLogoff", L"%SystemRoot%\\Media\\ReactOS_LogOff.wav" }
+/* Logon sound is already set by default for both Server and Workstation */
+};
+
+static const WCHAR* s_ExplorerSoundEvents[][2] = 
+{
+    { L"EmptyRecycleBin", L"%SystemRoot%\\Media\\ReactOS_Recycle.wav" },
+    { L"Navigating", L"%SystemRoot%\\Media\\ReactOS_Start.wav" }
+};
+
+static BOOL
+DoWriteSoundEvents(HKEY hKey,
+                   LPCWSTR lpSubkey,
+                   LPCWSTR lpValue[],
+                   DWORD dwSize)
+{
+    HKEY hRootKey, hEventKey, hDefaultKey;
+    LONG error;
+    ULONG i;
+    LPCWSTR pszData, pszDest[MAX_PATH][2];
+    DWORD cbData;
+
+    for (i = 0; i < dwSize; i++)
+    {
+        /* Expand environment variable from string */
+        if (!ExpandEnvironmentStringsW(&lpValue[i][1], (LPWSTR)pszDest, _countof(pszDest)))
+        {
+            /* Failed to expand, continue with the other files */
+            continue;
+        }
+
+        /* Check if the sound file exists and isn't a directory */
+        if (GetFileAttributesW(pszDest[i][1]) == INVALID_FILE_ATTRIBUTES ||
+            GetFileAttributesW(pszDest[i][1]) == FILE_ATTRIBUTE_DIRECTORY)
+        {
+            /* It does not, just continue with the other files */
+            continue;
+        }
+    }
+
+    /* Open sound events key */
+    error = RegOpenKeyExW(hKey, lpSubkey, 0, KEY_READ, &hRootKey);
+    if (error)
+    {
+        DPRINT1("RegOpenKeyExW failed\n");
+        goto Error;
+    }
+
+    for (i = 0; i < dwSize; i++)
+    {
+        /* Open all sound event subkeys */
+        error = RegOpenKeyExW(hRootKey, &lpValue[i][0], 0, KEY_READ, &hEventKey);
+        if (error)
+        {
+            DPRINT1("RegOpenKeyExW failed\n");
+            goto Error;
+        }
+
+        /* Open .Default subkey */
+        error = RegOpenKeyExW(hEventKey, L".Default", 0, KEY_WRITE, &hDefaultKey);
```

Comment:
Indeed @HBelusca. But if the key(s) will be created in any case, then do file existing checks above still make any sense? Will they still affect this behaviour? :thinking: I'm a bit confused.

Reply:
They don't seem to exclude each other... If the sound file for this event, that is specified in the list above, doesn't exist/is not a file in the installation media, then I don't know... You may choose to not create the sound event entry in the registry, or on the contrary, to create it in any case, but just keeping it empty...

Reply:
OK, thanks.
Since the situation with missing sound files is actual for now, it indeed makes sense.
I think it will be better just to open the keys without creating, since the sound events keys itself (and their `.Default` subkeys) are already existing in registry as well, same as in Windows (see our bootdata). So it does not make any sense to try create them in this case.
Another more important thing is, the **values** are empty by default. So the main choice is write the new ones or not. Is it possible to write them only if the sound files exist? :thinking: